### DISCUSSION TITLE
Skip response validation if BaseModel provided

### DISCUSSION THREAD

**Discussion by spumer** [fastapi/fastapi#8930] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * Or, I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Problem

FastAPI force validation in `serialize_response` even when validated model provided, by calling 

https://github.com/tiangolo/fastapi/blob/dc5a966548df973d0fd2d3a28bad65d0436f6fb9/fastapi/routing.py#L110

and then we convert BaseModel again to dict

https://github.com/tiangolo/fastapi/blob/dc5a966548df973d0fd2d3a28bad65d0436f6fb9/fastapi/routing.py#L66

which will be validated in next steps inside`serialize_response`

Twice model validation looks like wasting resources.

Also Pydantic allow creating model without validation: https://pydantic-docs.helpmanual.io/usage/models/#creating-models-without-validation and invalid models can't be returned as response (may be this ok, idk)

### Expected behaviour

When i return list/dict it will be validated trough `response_model`
When i return `response_model` instance no validation applied

### Example

```Python
from typing import Dict
from typing import Literal

import uvicorn
from pydantic import BaseModel
from fastapi import FastAPI

app = FastAPI()


class MyModel(BaseModel):
    msg: Dict[Literal['Hello'], Literal['World']]


@app.get("/revalidate", response_model=MyModel)
def revalidate():
    return {'msg': {"Hello": "World"}}


@app.get("/no_revalidate", response_model=MyModel)
def no_revalidate():
    return MyModel(msg={"Hello": "World"})


if __name__ == '__main__':
    uvicorn.run(app)
```

**Comment by falkben** [comment]
There's this PR that I think would resolve your issue: https://github.com/tiangolo/fastapi/pull/2347/

**Comment by spumer** [comment]
> There's this PR that I think would resolve your issue: #2347

Actually not, cause it's just use existing mechanic with custom Response

I can do that manually. But i need wrap my result model with ModelResponse each time, for all methods as suggested in #2347
I think we can do that more transparently

We also use fastapi-jsonrpc, and fix in basic serialization mechanic will automatically improve performance [in json-rpc serialization](https://github.com/smagafurov/fastapi-jsonrpc/blob/4ec510be3862ea532204fea376953d1c64c8ac65/fastapi_jsonrpc/__init__.py#L816)

**Comment by spumer** [comment]
To be clarify we explicit wraps result to `response_model`. This allow us use type-hinting to ensure all things are right ;)

**Comment by spumer** [comment]
This also correlates with https://github.com/tiangolo/fastapi/issues/2296

If we will use `return type annotation` instead `response_model` we should return prepared model instead ModelResponse (as suggested in #2347)
